src/config.rs

```diff
@@ -120,12 +135,21 @@ pub fn config_path() -> Result<PathBuf> {
 ///
 /// - Fails if the home directory cannot be found
 /// - Fails if the config directory cannot be created
-pub fn save_config(config: &Config) -> Result<()> {
-    let config_path = config_path()?;
-    let mut file = File::create(config_path)?;
-    let config = toml::to_string(config)?;
-    file.write_all(config.as_bytes())?;
-    Ok(())
+
+pub fn save_config(config: &Config) -> Result<(), ConfigError> {
+    // Tries to catch the config path
+    let config_path = config_path().map_err(|_| ConfigError::CreateConfigDir)?;
+    // Tries to create the config file
+    let mut file = File::create(&config_path).map_err(|_| ConfigError::CreateConfigFile)?;
```

Comment:
This returns a `std::io::Error`, see https://doc.rust-lang.org/std/fs/struct.File.html#method.create

Can you use the `#[from]` stuff in `thiserror` to convert `std::io::Error` to `ConfigError::CreateConfigFile`? I've added a comment above as well.
Here should be:

```suggestion
    let mut file = File::create(&config_path)?;
```